server/controllers/favorites.js

```diff
@@ -1,38 +1,95 @@
 import { Router } from 'express'
 import Favorite from '../models/Favorite.js'
 import authenticate from '../middleware/authenticate.js'
-import authorize from '../middleware/authorize.js'
 
-const ONE_DAY_IN_MS = 1000 * 60 * 60 * 24;
+
+const COINGECKO_API = 'https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=100&page=1&sparkline=false';
 
 export default Router()
-    
 
-    .get('/favorites', async (req, res, next) => {
+    .get('/', authenticate,async (req, res, next) => {
         try {
-            const favorites = await FavoriteService.getAllFavorites();
-            res.send(favorites);
+            const favorites = await Favorite.getAll()
+            res.status(200).json(favorites);
         } catch (error) {
             next(error);
         }
     })
-    .post('/:id/favorites', async (req, res, next) => {
+
+    .get('/userfavorites',authenticate, async (req, res, next) => {
         try {
-            const favorite = await FavoriteService.addFavorite(req.body);
-            res.json(favorite);
-        } catch (e) {
-            next(e);
+            const favorites = await Favorite.getUserFavorites(req.user.id)
+            res.status(200).json(favorites);
+        } catch (error) {
+            next(error);
         }
     })
     
+    .get('/coins/userfavorites',authenticate, async (req, res, next) => {
+        try {
+            const favorites = await Favorite.getUserFavorites(req.user.id)
+
+            if(!favorites.coins_ids) return res.status(200).json([])
+
+            const coins_res = await fetch(`${COINGECKO_API}&ids=${favorites.coins_ids}`)
+            const coins_json = await coins_res.json()
+
+            res.status(200).json(coins_json);
+
+        } catch (error) {
+            next(error);
+        }
+    })
+
     
-    .delete('/favorites/:id', async (req, res, next) => {
+
+    .post('/add/userfavorites',authenticate, async (req, res, next) => {
+        const { coin_id } = req.body
         try {
-            const favorite = await FavoriteService.deleteFavorite(req.params.id);
-            res.json(favorite);
-        } catch (e) {
-            next(e);
+            let { coins_ids } = await Favorite.getUserFavorites(req.user.id)
+            
```

Comment:
If having performance issues for large amounts of coins later on perhaps implement a for loop.